File: 프로그래머스/unrated/181882. 조건에 맞게 수열 변환하기 1/조건에 맞게 수열 변환하기 1.js
function solution(arr) {
    return arr.map((el, i) => {
        if (el >= 50 && el % 2 === 0) {
            return el / 2;
        } else if (el < 50 && el % 2 === 1) {
            return el * 2;
        } else {
            return el;
        }
    });
}

File: 프로그래머스/unrated/181882. 조건에 맞게 수열 변환하기 1/조건에 맞게 수열 변환하기 1.java
class Solution {
    public int[] solution(int[] arr) {
        for (int i = 0; i < arr.length; i++) {
            boolean isEven = arr[i] % 2 == 0;
            boolean isLargerOrEqualToFifty = arr[i] >= 50;
            
            if (isEven && isLargerOrEqualToFifty) {
                arr[i] /= 2;
            } else if (!isEven && !isLargerOrEqualToFifty) {
                arr[i] *= 2;
            }
        }
        
        return arr;
    }
}